amplify/function/BusinessLogic/ResetHackathon/handler.ts

```diff
@@ -0,0 +1,267 @@
+import {
+  deleteScore,
+  deleteTeam,
+  deleteTeamRoom,
+  deleteUser,
+  updateHackathon,
+} from "./graphql/mutations";
+import {
+  listHackathons,
+  listScores,
+  listTeamRooms,
+  listTeams,
+  listUsers,
+} from "./graphql/queries";
+
+Amplify.configure(
+  {
+    API: {
+      GraphQL: {
+        endpoint: process.env.AMPLIFY_DATA_GRAPHQL_ENDPOINT as string,
+        region: process.env.AWS_REGION,
+        defaultAuthMode: "iam",
+      },
+    },
+  },
+  {
+    Auth: {
+      credentialsProvider: {
+        getCredentialsAndIdentityId: async () => ({
+          credentials: {
+            accessKeyId: process.env.AWS_ACCESS_KEY_ID as string,
+            secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY as string,
+            sessionToken: process.env.AWS_SESSION_TOKEN as string,
+          },
+        }),
+        clearCredentialsAndIdentityId: () => {
+          /* noop */
+        },
+      },
+    },
+  },
+);
+
+const client = generateClient<Schema>({
+  authMode: "iam",
+});
+
+type Handler = Schema["ResetHackathon"]["functionHandler"];
+
+export const handler: Handler = async (event) => {
+  try {
+    // Extract arguments from the event if any
+    const {
+      safetyCheck,
+      resetRooms,
+      resetScores,
+      resetTeams,
+      resetUsers,
+      startDate,
+      endDate,
+      scoreComponents,
+      scoringSidepots,
+    } = event.arguments;
+    if (safetyCheck !== process.env.EDIT_HACKATHON_CODE) {
+      return {
+        statusCode: 403,
+        headers: { "Content-Type": "application/json" },
+      };
+    }
+    // get the current hackathon model data items
+    const HackathonItems = (
+      await client.graphql({
+        query: listHackathons,
+      })
+    ).data.listHackathons.items;
+
+    // Assuming there is only 1 Hackathon
+    if (HackathonItems.length === 0) {
+      console.log("Not a single Hackathon, please create one");
+      return {
+        statusCode: 400,
+        headers: { "Content-Type": "application/json" },
+      };
+    }
+    // Assuming the first Hackathon is the right one
+    const HackathonID = HackathonItems[0].id;
+
+    // Resetting Users variable
+    const resettingUsers: boolean = resetUsers ?? false;
+    // Reset teams if users are being reset
+    const resettingTeams: boolean = (resetTeams ?? false) || resettingUsers;
+    // Reset Rooms if teams are being reset
+    const resettingRooms: boolean = (resetRooms ?? false) || resettingTeams;
+    // Reset scores if teams are being reset
+    const resettingScores: boolean = (resetScores ?? false) || resettingTeams;
+
+    // Reset Users
+    if (resettingUsers) {
+      console.log("resetting all users");
+      const usersResponse = await client.graphql({
+        query: listUsers,
+      });
+      const users = usersResponse.data.listUsers.items;
+      for (const user of users) {
+        const id = user.id;
+        // only delete the participants
+        if (user.role === "Participant") {
```

Comment:
Looks great, we also need to ensure we delete the users from Cognito. Look at the AddUserToGroup lambda function to see how to do that. You should only need the `AdminDeleteUserCommand`. Check the auth/resource file to see how to grant this lambda permission to Cognito Auth similar to how we've granted permission to data.  Search the repo for this for what I mean: 
```allow
      .resource(AddUserToGroup)
      .to(["addUserToGroup", "removeUserFromGroup", "listGroupsForUser"]),```